File: orders/src/main/java/com/github/orders/dto/OrderDto.java
package com.github.orders.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.orders.entity.OrderStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderDto {

    @JsonProperty(value = "id")
    private Long id;

    @NotNull
    @JsonProperty(value = "customer")
    private CustomerDto customer;

    @NotEmpty
    @JsonProperty(value = "products")
    private List<ProductDto> products;

    @NotNull
    @JsonProperty(value = "amount")
    private BigDecimal amount;

    @NotNull
    @JsonProperty(value = "status")
    private OrderStatus status;

    @NotNull
    @JsonProperty(value = "delivery")
    private DeliveryDto delivery;

}

File: managers/src/main/go/com.github/dto/order_dto.go
package dto

// OrderDto swagger:model
type OrderDto struct {
	Id       uint         `json:"id" example:"1"`
	Customer CustomerDto  `json:"customer"`
	Products []ProductDto `json:"product"`
	Amount   float64      `json:"amount" example:"1.99"`
	Status   string       `json:"status" example:"status"`
	Delivery DeliveryDto  `json:"delivery"`
}
